src/python_testing/post_certification_tests/post-cert-checks.py

```diff
@@ -0,0 +1,348 @@
+
+
+def get_dcl_compliance_info(vid, pid, software_version):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/compliance/compliance-info/{vid}/{pid}/{software_version}/matter").json()
+
+
+def get_dcl_certified_model(vid, pid, software_version):
+    return requests.get(f"{fetch_paa_certs_from_dcl.PRODUCTION_NODE_URL_REST}/dcl/compliance/certified-models/{vid}/{pid}/{software_version}/matter").json()
+
+
+class DclCheck(MatterBaseTest, BasicCompositionTests):
+    @async_test_body
+    async def setup_class(self):
+        self.connect_over_pase(self.default_controller)
+        bi = Clusters.BasicInformation
+        self.vid = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.VendorID)
+        self.pid = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.ProductID)
+        self.software_version = await self.read_single_attribute_check_success(cluster=bi, attribute=bi.Attributes.SoftwareVersion)
+
+    def steps_Vendor(self):
+        return [TestStep(1, "Check if device VID is listed in the DCL vendor schema", "Listing found")]
+
+    def test_Vendor(self):
+        self.step(1)
+        entry = get_dcl_vendor(self.vid)
+        key = 'vendorInfo'
+        asserts.assert_true(key in entry.keys(), f"Unable to find vendor entry for {self.vid:04x}")
+        logging.info(f'Found vendor key 0x{self.vid:04X} in DCL:')
+        logging.info(f'{entry[key]}')
+
+    def steps_Model(self):
+        return [TestStep(1, "Check if device VID/PID are listed in the DCL model schema", "Listing found")]
+
+    def test_Model(self):
+        self.step(1)
+        key = 'model'
+        entry = get_dcl_model(self.vid, self.pid)
+        asserts.assert_true(key in entry.keys(), f"Unable to find model entry for {self.vid:04x} {self.pid:04x}")
+        logging.info(f'Found model entry for vid=0x{self.vid:04X} pid=0x{self.pid:04X} in the DCL:')
+        logging.info(f'{entry[key]}')
+
+    def steps_Compliance(self):
+        return [TestStep(1, "Check if device VID/PID/SoftwareVersion are listed in the DCL compliance info schema", "Listing found")]
+
+    def test_Compliance(self):
+        self.step(1)
+        key = 'complianceInfo'
+        entry = get_dcl_compliance_info(self.vid, self.pid, self.software_version)
+        asserts.assert_true(key in entry.keys(),
+                            f"Unable to find compliance entry for {self.vid:04x} {self.pid:04x} {self.software_version}")
+        logging.info(
+            f'Found compliance info for vid=0x{self.vid:04X} pid=0x{self.pid:04X} software version={self.software_version} in the DCL:')
+        logging.info(f'{entry[key]}')
+
+    def steps_CertifiedModel(self):
+        return [TestStep(1, "Check if device VID/PID/SoftwareVersion are listed in the DCL certified model schema", "Listing found")]
+
+    def test_CertifiedModel(self):
```

Comment:
We've seen a few cases where a software update is posted in the DCL and the `OtaChecksum` is not base64 encoded. What do you think about adding a test that verifies at least that all the software versions for this device model have a correctly formatted checksum? Could potentially also verify other fields as well but this one in particular feels especially important.

Reply:
Yeah, that seems like a good addition. Let me see how to go about doing that - right now I'm just pulling the entry for this specific device version, and I'm not totally sure how to get the index. Give me like a day or two.

Reply:
I also wonder if this is something that the DCL should be checking before accepting a new entry....

@jcps07 - do you know if it would be possible to pre-check entries for this as they're uploaded?

Reply:
I'll bring it up to the DCL-TT and create an issue.

Reply:
awesome - thanks

Reply:
[Issue #570](https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/570) created

Reply:
OK added a more complete check where it verifies the OTA image fully. PTAL if you have a sec.